components/builder-db/src/models/account.rs

```diff
@@ -1,77 +1,100 @@
+    pub created_at: Option<NaiveDateTime>,
+    pub updated_at: Option<NaiveDateTime>,
 }
 
-pub struct GetAccount {
-    pub name: String,
-}
-
-pub struct GetAccountById {
-    pub id: i64,
-}
-
-pub struct CreateAccount {
-    pub name: String,
-    pub email: String,
-}
-
-pub struct UpdateAccount {
+#[derive(Identifiable, Debug, Serialize, QueryableByName)]
+#[table_name = "account_tokens"]
+pub struct AccountToken {
+    #[serde(with = "db_id_format")]
     pub id: i64,
-    pub email: String,
+    #[serde(with = "db_id_format")]
+    pub account_id: i64,
+    pub token: String,
+    pub created_at: Option<NaiveDateTime>,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-pub struct FindOrCreateAccount {
-    pub name: String,
-    pub email: String,
+#[derive(Insertable)]
+#[table_name = "accounts"]
+pub struct NewAccount<'a> {
+    pub email: &'a str,
+    pub name: &'a str,
 }
 
 impl Account {
-    pub fn get(account: GetAccount, conn: &PgConnection) -> QueryResult<Account> {
+    pub fn get(name: &str, conn: &PgConnection) -> QueryResult<Account> {
         diesel::sql_query("SELECT * FROM get_account_by_name_v1($1)")
-            .bind::<Text, _>(account.name)
+            .bind::<Text, _>(name)
             .get_result(conn)
     }
 
-    pub fn get_by_id(account: GetAccountById, conn: &PgConnection) -> QueryResult<Account> {
+    pub fn get_by_id(id: u64, conn: &PgConnection) -> QueryResult<Account> {
         diesel::sql_query("SELECT * FROM get_account_by_id_v1($1)")
-            .bind::<BigInt, _>(account.id)
+            .bind::<BigInt, _>(id as i64)
             .get_result(conn)
     }
 
-    pub fn create(account: CreateAccount, conn: &PgConnection) -> QueryResult<Account> {
+    pub fn create(account: &NewAccount, conn: &PgConnection) -> QueryResult<Account> {
         diesel::sql_query("SELECT * FROM select_or_insert_account_v1($1, $2)")
             .bind::<Text, _>(account.name)
             .bind::<Text, _>(account.email)
             .get_result(conn)
     }
 
-    pub fn find_or_create(
-        account: FindOrCreateAccount,
-        conn: &PgConnection,
-    ) -> QueryResult<Account> {
+    pub fn find_or_create(name: &str, email: &str, conn: &PgConnection) -> QueryResult<Account> {
         diesel::sql_query("SELECT * FROM select_or_insert_account_v1($1, $2)")
-            .bind::<Text, _>(account.name)
-            .bind::<Text, _>(account.email)
+            .bind::<Text, _>(name)
+            .bind::<Text, _>(email)
             .get_result(conn)
     }
 
-    pub fn update(account: UpdateAccount, conn: &PgConnection) -> QueryResult<usize> {
+    pub fn update(id: u64, email: &str, conn: &PgConnection) -> QueryResult<usize> {
         diesel::sql_query("SELECT update_account_v1($1, $2)")
-            .bind::<BigInt, _>(account.id)
-            .bind::<Text, _>(account.email)
+            .bind::<BigInt, _>(id as i64)
+            .bind::<Text, _>(email)
+            .execute(conn)
+    }
+}
+
+#[derive(Insertable)]
```

Comment:
Do you need to derive insertable here if we're not using the query dsl?

Reply:
I'm leaving those in mostly as a placeholder for experimenting with direct inserts